tck/src/main/java/org/apache/jdo/tck/pc/mylib/PCRect2.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jdo.tck.pc.mylib;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import javax.jdo.AttributeConverter;
+import java.io.Serializable;
+import java.util.Date;
+
+/**
+ * PersistenceCapable class to test JDO AttributeConverter interface.
+ * It#s fields of type Point are converted to strins in the datastore.
+ *
+ * @author Michael Bouschen
+ */
+public class PCRect2 {
+    private static long counter = new Date().getTime();
+
+    private static long newId() {
+        synchronized (PCRect2.class) {
+            return counter++;
+        }
+    }
+
+    public long id = newId();
+
+    private Point upperLeft;
+    private Point lowerRight;
+
+    public PCRect2() {
+    }
+
+    public PCRect2(long id, Point ul, Point lr) {
+        upperLeft = ul;
+        lowerRight = lr;
+    }
+
+    public PCRect2(Point ul, Point lr) {
+        upperLeft = ul;
+        lowerRight = lr;
+    }
+
+    public Point getUpperLeft() {
+        return upperLeft;
+    }
+
+    public Point getLowerRight() {
+        return lowerRight;
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public void setId(long id) {
+        this.id = id;
+    }
+
+    public String toString() {
+        String rc = null;
+        Object obj = this;
+        try {
+            //rc = Util.getClassName(this)
+            rc = obj.getClass().getName()
+                    + " ul: " + getUpperLeft().name()
+                    + " lr: " + getLowerRight().name();
+        } catch (NullPointerException ex) {
+            rc = "NPE getting PCRect2's values";
+        }
+        return rc;
+    }
+
+    /**
+     * PCRect2'S ObjectId class.
+     */
+    public static class Oid implements Serializable {
```

Comment:
Do we need this `Oid` class?

Reply:
We need an Oid class for applicationidentity.

Reply:
I replaced the Oid class by using SingleFieldIdentity. Now the RCRectString metadata defines  javax.jdo.identity.LongIdentity as ObjectId class.